pkg/onchain/sc.go

```diff
@@ -77,6 +77,8 @@ func CallFunctionUnwaited(client *node.Client, wallet wallet.Wallet, expiryDelta
 	return operationID, nil
 }
 
+// DeploySC deploys a smart contract on the blockchain. It returns the address of the smart contract and an Error.
+// The smart contract is deployed with the given wallet.
 func DeploySC(client *node.Client, wallet wallet.Wallet, contract []byte) (string, error) {
```

Comment:
Ok so yeah, this won't do it. We are supposed to use the plugin wallet. The local wallet is deprecated. This means here you should pass a wallet nickname instead.